examples/unsloth/unsloth.py

```diff
@@ -0,0 +1,66 @@
+# Use the unsloth library to fine-tune a Mistral model
+
+from datasets import load_dataset
+import torch
+from transformers import TrainingArguments
+from trl import SFTTrainer
+from unsloth import FastLanguageModel
+
+max_seq_length = 2048
+
+# [1] Get LAION dataset
+url = "https://huggingface.co/datasets/laion/OIG/resolve/main/unified_chip2.jsonl"
+dataset = load_dataset("json", data_files = {"train" : url}, split = "train")
+
+fourbit_models = [
+    "unsloth/mistral-7b-bnb-4bit",
+    "unsloth/llama-2-7b-bnb-4bit",
+    "unsloth/llama-2-13b-bnb-4bit",
+    "unsloth/codellama-34b-bnb-4bit",
+    "unsloth/tinyllama-bnb-4bit",
+]
+
+# [2] Load Mistral model
+model, tokenizer = FastLanguageModel.from_pretrained(
+    model_name = "unsloth/mistral-7b-bnb-4bit",
+    max_seq_length = max_seq_length,
+    dtype = None,
+    load_in_4bit = True,
+)
+
+# [3] Do model patching and add fast LoRA weights
+model = FastLanguageModel.get_peft_model(
+    model,
+    r = 16,
+    target_modules = ["q_proj", "k_proj", "v_proj", "o_proj",
+                      "gate_proj", "up_proj", "down_proj",],
+    lora_alpha = 16,
+    lora_dropout = 0,
+    bias = "none",
+    use_gradient_checkpointing = True,
+    random_state = 3407,
+    max_seq_length = max_seq_length,
+)
+
+# [4] Initialize and train the model using the SFTTrainer
+trainer = SFTTrainer(
+    model = model,
+    train_dataset = dataset,
+    dataset_text_field = "text",
+    max_seq_length = max_seq_length,
+    tokenizer = tokenizer,
+    args = TrainingArguments(
+        per_device_train_batch_size = 2,
+        gradient_accumulation_steps = 4,
+        warmup_steps = 10,
+        max_steps = 60,
+        fp16 = not torch.cuda.is_bf16_supported(),
+        bf16 = torch.cuda.is_bf16_supported(),
+        logging_steps = 1,
+        output_dir = "outputs",
```

Comment:
Can we set output_dir as an arg to this script (e.g., `python unsloth.py --output-dir /outputs`)?  Consider using argparse for this.

Then we can add this to unsloth.yaml to showcase the storage access abilities of SkyPilot:

```
file_mounts:
  /outputs:
    name: my-unsloth-checkpoints

...

run: |
  python unsloth.py --output-dir /outputs  # Checkpoints will be written directly to object store
```

Reply:
i'm on it!

Reply:
done